server/src/main/java/io/pravega/schemaregistry/storage/client/WireCommandClient.java

```diff
@@ -0,0 +1,417 @@
+                    .put(WireCommands.ReadTableEntries.class, ImmutableSet.of(WireCommands.TableEntriesRead.class))
+                    .build();
+
+    private static final Map<Class<? extends Request>, Set<Class<? extends Reply>>> EXPECTED_FAILING_REPLIES =
+            ImmutableMap.<Class<? extends Request>, Set<Class<? extends Reply>>>builder()
+                    .put(WireCommands.UpdateTableEntries.class, ImmutableSet.of(WireCommands.TableKeyDoesNotExist.class,
+                            WireCommands.TableKeyBadVersion.class, WireCommands.NoSuchSegment.class))
+                    .put(WireCommands.RemoveTableKeys.class, ImmutableSet.of(WireCommands.TableKeyBadVersion.class, WireCommands.NoSuchSegment.class))
+                    .put(WireCommands.DeleteTableSegment.class, ImmutableSet.of(WireCommands.TableSegmentNotEmpty.class))
+                    .put(WireCommands.ReadTable.class, ImmutableSet.of(WireCommands.NoSuchSegment.class))
+                    .put(WireCommands.ReadTableKeys.class, ImmutableSet.of(WireCommands.NoSuchSegment.class))
+                    .put(WireCommands.ReadTableEntries.class, ImmutableSet.of(WireCommands.NoSuchSegment.class))
+                    .build();
+
+    @Getter(AccessLevel.PACKAGE)
+    private final HostStore hostStore;
+    private final ConnectionPool connectionPool;
+
+    WireCommandClient(final ConnectionPool connectionPool, HostStore hostStore) {
+        this.connectionPool = connectionPool;
+        this.hostStore = hostStore;
+    }
+
+    CompletableFuture<NodeUri> getTableUri(final String tableName) {
+        return hostStore.getHostForTableSegment(tableName);
+    }
+
+    /**
+     * This method sends a WireCommand to create a table segment.
+     *
+     * @param tableName           Qualified table name.
+     * @param delegationToken     The token to be presented to the segmentstore.
+     * @return A CompletableFuture that, when completed normally, will indicate the table segment creation completed
+     * successfully. If the operation failed, the future will be failed with the causing exception. If the exception
+     * can be retried then the future will be failed with {@link StoreExceptions}.
+     */
+    CompletableFuture<Void> createTableSegment(final String tableName,
+                                               String delegationToken) {
+        return getTableUri(tableName)
+                .thenCompose(uri -> {
+                    final WireCommandType type = WireCommandType.CREATE_TABLE_SEGMENT;
+
+                    RawClient connection = new RawClient(ModelHelper.encode(uri), connectionPool);
+                    final long requestId = connection.getFlow().asLong();
+
+                    return sendRequest(connection, requestId, new WireCommands.CreateTableSegment(requestId, tableName, false, delegationToken))
+                            .thenAccept(rpl -> handleReply(rpl, connection, tableName, WireCommands.CreateTableSegment.class, type));
```

Comment:
I suppose there can be exceptions here and they should be handled too.

Reply:
sendrequest method actually handles the exceptions. so we dont need exception handling in each of the indivudual methods